videos/utils.py

```diff
@@ -46,3 +50,63 @@ def get_content_dirpath(slug, collection_type):
             continue
         return collection["folder"]
     return None
+
+
+def copy_obj_s3(source_obj: WebsiteContent, dest_course: Website) -> str:
+    """Copy source_obj to the S3 bucket of dest_course"""
+    s3 = get_boto3_resource("s3")
+    new_s3_path = generate_s3_path(source_obj, dest_course)
+    s3.Object(settings.AWS_STORAGE_BUCKET_NAME, new_s3_path).copy_from(
+        CopySource=f"{settings.AWS_STORAGE_BUCKET_NAME.rstrip('/')}/{str(source_obj.file).lstrip('/')}"
+    )
+    return new_s3_path
+
+
+def create_new_content(source_obj, to_course):
+    """Create new WebsiteContent object from source_obj in to_course,
+    or update existing object if it exists.
+    """
+    new_text_id = uuid_string()
+    if source_obj.file:
+        new_s3_loc = copy_obj_s3(source_obj, to_course)
+        new_dirpath = get_dirpath_and_filename(new_s3_loc)[0]
```

Comment:
There seems to be an issue here with the `dirpath` attribute. Take a look at my test course that I've been working with:

https://github.com/gumaerc-testorg/video-copy-test-site/tree/main

As you can see, there is a `courses/video-copy-test-site` folder in the Github repo that shouldn't be there. I tried deleting my website and recreating it, and the problem persisted. The `dirpath` property controls where the markdown representing a piece of content is stored. In this case, these files should be under `content/resources`.

Reply:
As a separate note, if you look at the content in that folder you'll see `file` paths like `/ol-ocw-studio-app/courses/video-copy-test-site/-EgKluVR2Ug_captions.vtt`. This may be related to https://github.com/mitodl/ocw-studio/issues/2117 but we should check and make sure the correct `file` attribute is being written here.